### ISSUE TITLE
AGD Carousel for AtoMMC bug

### ISSUE THREAD

**Issue by oss003** [hoglet67/AtomSoftwareArchive#3]
In the AtoMMC version of Carousel, there is a *ROFF command.
Also the Carousel for AtoMMC is set to Econet (E=1)

**Comment by hoglet67** [comment]
The *ROFF command is part of the logic for detection Econet vs AtoMMC.

Can you confirm what version of the archive you are running?

Does your Atom have a *ROFF command?

**Comment by hoglet67** [comment]
The detection code is:
```
    11 DIM C 100
    12 REM: DETECT ECONET
    13 E=0
    14 Z=!16
    15 $C="GOTO 20"
    16 ?16=C;?17=C&#FFFF/256
    17 REM: ISSUE ECONET-ONLY COMMAND
    18 *ROFF
    19 E=1
    20 ?16=Z;?17=Z&#FFFF/256
```
If a *ROFF command is present, then E=1 else E=0.

I've just tested this in Atomulator and it works.

What exactly is happening in your case?

**Comment by hoglet67** [comment]
This pattern is basically equivalent to:
```
ON ERROR GOTO 20
```

**Comment by oss003** [comment]
![image](https://user-images.githubusercontent.com/4426956/233476613-62212001-08d9-4685-9ef4-0b33d4552ac1.png)

**Comment by hoglet67** [comment]
It looks like the "ON ERROR" code I'm using is not catching the error in your case.

What version of AtoMMC are you using?

Dave

**Comment by oss003** [comment]
![image](https://user-images.githubusercontent.com/4426956/233477747-b638f67c-b4c9-40d9-a654-99c4e550d781.png)

**Comment by hoglet67** [comment]
Ah, that's probably this issue.,

AtoMMC3 errors result in a BRK which can be caught.

AtoMMC2 errors end up exiting back to BASIC I think.

Can you think of a better way of detecting AtoMMC vs Econet that would work with all versions?

**Comment by oss003** [comment]
Look at the hardware addresses?
AtoMMC has a heartbeat.

**Comment by oss003** [comment]
Econet sets vector addresses to #0239

**Comment by hoglet67** [comment]
I might test for the IRQ vector pointing to a BIT &B4xx:
```
.irq_handler
    bit reg_adlc_status1                                              ; a0dc: 2c 00 b4    ,..
```
So something like:
```
E=0
A=!#204&#FFFF
IF ?A=#2C AND A?2=#B4 E=1
```
That would be very unlikely to generate a false positive with AtoMMC. And I don't think it's likely that we will re-write the Econet IRQ handler!

I think this is better than testing #238/9 as when switching from Econet back to AtoMMC that "signature" would remain.

**Comment by hoglet67** [comment]
Fixed in V13.00
